flink-runtime/src/test/java/org/apache/flink/runtime/state/StateBackendTestBase.java

```diff
@@ -193,24 +193,24 @@ protected CheckpointStreamFactory createStreamFactory() throws Exception {
         return checkpointStreamFactory;
     }
 
-    protected <K> AbstractKeyedStateBackend<K> createKeyedBackend(TypeSerializer<K> keySerializer)
-            throws Exception {
+    protected <K> CheckpointableKeyedStateBackend<K> createKeyedBackend(
```

Comment:
Actually, I don't know why we have to change the returned class from `AbstractKeyedStateBackend` to `CheckpointableKeyedStateBackend`, is it necessary?

Reply:
The reason is that all the tests (including 4 different extensions of StateBackendTestBase.java) are reused by ChangelogStateBackend. 

"ChangelogKeyedStateBackend" is not an "AbstractKeyedStateBackend"; it is a `CheckpointableKeyedStateBackend` instead, that's why the change is made.

If not made this way, we would have to rewrite and duplicatete all the tests (which are similar to what are included in StateBackendTestBase)